### PR TITLE
Ntueeplus-260 create DB and backend API for study abroad sharing session

### PR DESCRIPTION
**Pull request by weiber2002**
### What is this PR for?

Make a DB model and backend api for crud data of study abroad sharing session
### What type of PR is it?
Feature

### Todos
* [x] - Task
CRUD of abroad sharing session

### What is the Jira issue?

https://ntueeplus.atlassian.net/browse/NTUEEPLUS-260

### CODE REVIEW

File: backend/routes/srcs/in/abroadSharing/addAbroadSharing.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * @apiError (500) {String} description 資料庫錯誤
+ */
+const addAbroadSharing = async (req, res) => {
+  const { intro, YTlink, otherLinks } = req.body
+  const { _id } = await new AbroadSharing({ intro, YTlink, otherLinks }).save().catch(dbCatch)
+  res.status(201).send({ _id })
+}
+
+const valid = require('../../../middleware/validation')
+const rules = [{ filename: 'optional', field: ['intro', 'YTlink', 'otherLinks'], type: 'String' }]
```
Comment (vwvwMM) [inline backend/routes/srcs/in/abroadSharing/addAbroadSharing.js:26]: YTLink and otherLinks should use type url

File: backend/routes/srcs/in/abroadSharing/getAbroadSharing.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ *
+ * @apiSuccess (201) {String} intro  介紹
+ * @apiSuccess (201) {String} YTlink youtube連結
+ * @apiSuccess (201) {String} otherLinks 其他連結
+ *
+ * @apiError (500) {String} description 資料庫錯誤
+ */
+const getAbroadSharing = async (req, res, next) => {
+  const sharing = await abroad_sharing.find({}).catch(dbCatch)
+  res.status(201).send(sharing)
```
Comment (vwvwMM) [inline backend/routes/srcs/in/abroadSharing/getAbroadSharing.js:18]: if not found, please return 404

File: backend/routes/srcs/in/abroadSharing/main.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+const express = require('express')
+const router_auth = express.Router()
+const router = express.Router()
+
+router_auth.post('/addAbroadSharing', require('./addAbroadSharing'))
+router.get('/getAbroadSharing', require('./getAbroadSharing'))
+router_auth.delete('/deleteAbroadSharing', require('./deleteAbroadSharing'))
+router_auth.patch('/updateAbroadSharing', require('./updateAbroadSharing'))
+module.exports = { router, router_auth }
```
Comment (vwvwMM) [inline backend/routes/srcs/in/abroadSharing/main.js:9]: split router and router_auth st. it is more clear which are for router and which are for router_auth

File: backend/routes/srcs/in/abroadSharing/updateAbroadSharing.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ * @apiSuccess (200) _id _id
+ *
+ * @apiError (404) {String} description 資料不存在
+ * @apiError (403) {String} description _id doesnot exist
+ * @apiError (500) {String} description 資料庫錯誤
+ */
+const updateAbroadSharing = async (req, res, next) => {
+  const { _id, intro, YTlink, otherLinks } = req.body
+  if (!_id) throw new ErrorHandler(403, '_id doesnot exist')
```
Comment (vwvwMM) [inline backend/routes/srcs/in/abroadSharing/updateAbroadSharing.js:26]: don't say "_id doesn't exist", use "please provide _id" may be better

**Comment by weiber2002** [comment]
add image  // type of link on api change to url // add title and intro

**Pull request closed without merging**